Source/Core/Core/HW/DSPHLE/UCodes/AXWiiOld.cpp

```diff
@@ -0,0 +1,190 @@
+      addr_lo = m_cmdlist[curr_idx++];
+      addr2_hi = m_cmdlist[curr_idx++];
+      addr2_lo = m_cmdlist[curr_idx++];
+      MixAUXSamples(cmd - CMD_MIX_AUXA, HILO_TO_32(addr), HILO_TO_32(addr2), volume);
+      break;
+
+    case CMD_UPL_AUXA_MIX_LRSC:
+    case CMD_UPL_AUXB_MIX_LRSC:
+    {
+      volume = m_cmdlist[curr_idx++];
+      u32 addresses[6] = {
+          (u32)(m_cmdlist[curr_idx + 0] << 16) | m_cmdlist[curr_idx + 1],
+          (u32)(m_cmdlist[curr_idx + 2] << 16) | m_cmdlist[curr_idx + 3],
+          (u32)(m_cmdlist[curr_idx + 4] << 16) | m_cmdlist[curr_idx + 5],
+          (u32)(m_cmdlist[curr_idx + 6] << 16) | m_cmdlist[curr_idx + 7],
+          (u32)(m_cmdlist[curr_idx + 8] << 16) | m_cmdlist[curr_idx + 9],
+          (u32)(m_cmdlist[curr_idx + 10] << 16) | m_cmdlist[curr_idx + 11],
+      };
+      curr_idx += 12;
+      UploadAUXMixLRSC(cmd == CMD_UPL_AUXB_MIX_LRSC, addresses, volume);
+      break;
+    }
+
+    // TODO(delroth): figure this one out, it's used by almost every
+    // game I've tested so far.
+    case CMD_UNK_0B:
+      curr_idx += 4;
+      break;
+
+    case CMD_OUTPUT:
+    case CMD_OUTPUT_DPL2:
+      addr_hi = m_cmdlist[curr_idx++];
+      addr_lo = m_cmdlist[curr_idx++];
+      addr2_hi = m_cmdlist[curr_idx++];
+      addr2_lo = m_cmdlist[curr_idx++];
+      OutputSamples(HILO_TO_32(addr2), HILO_TO_32(addr), 0x8000, cmd == CMD_OUTPUT_DPL2);
+      break;
+
+    case CMD_WM_OUTPUT:
+    {
+      u32 addresses[4] = {
+          (u32)(m_cmdlist[curr_idx + 0] << 16) | m_cmdlist[curr_idx + 1],
+          (u32)(m_cmdlist[curr_idx + 2] << 16) | m_cmdlist[curr_idx + 3],
+          (u32)(m_cmdlist[curr_idx + 4] << 16) | m_cmdlist[curr_idx + 5],
+          (u32)(m_cmdlist[curr_idx + 6] << 16) | m_cmdlist[curr_idx + 7],
+      };
+      curr_idx += 8;
+      OutputWMSamples(addresses);
+      break;
+    }
+
+    case CMD_END:
+      end = true;
+      break;
+    }
+  }
+}
+
+void AXWiiOldUCode::ProcessPBList(u32 pb_addr)
+{
+  // Samples per millisecond. In theory DSP sampling rate can be changed from
+  // 32KHz to 48KHz, but AX always process at 32KHz.
+  const u32 spms = 32;
+
+  AXPBWiiOld pb;
+
+  while (pb_addr)
+  {
+    AXBuffers buffers = {{m_samples_left,      m_samples_right,      m_samples_surround,
+                          m_samples_auxA_left, m_samples_auxA_right, m_samples_auxA_surround,
+                          m_samples_auxB_left, m_samples_auxB_right, m_samples_auxB_surround,
+                          m_samples_auxC_left, m_samples_auxC_right, m_samples_auxC_surround,
+                          m_samples_wm0,       m_samples_aux0,       m_samples_wm1,
+                          m_samples_aux1,      m_samples_wm2,        m_samples_aux2,
+                          m_samples_wm3,       m_samples_aux3}};
+
+    ReadPB(pb_addr, pb, m_crc);
+
+    u32 updates_addr = HILO_TO_32(pb.updates.data);
+    u16* updates = (u16*)HLEMemory_Get_Pointer(updates_addr);
```

Comment:
```suggestion
    u16* updates = static_cast<u16*>(HLEMemory_Get_Pointer(updates_addr));
```
(or a different kind of C++ cast; but I'm fairly certain that this should be allowed)

Reply:
You're not wrong, but there's so much code in here that's using C-style casts that it seems a bit silly to change just one.

Reply:
Baby steps I suppose...